app/models/UserAnswers.scala

```diff
@@ -37,7 +37,7 @@ final case class UserAnswers(
   def getPageValue[A](page: Gettable[A])(implicit rds: Reads[A]): EitherNec[ValidationError, A] =
     get(page).map(Right(_)).getOrElse(Left(NonEmptyChain.one(PageMissing(page))))
 
-  def getOptionalPageValue(
+  def getOptionalPageStringValue(
```

Comment:
Can do this with typing and so not need to define the Int one e.g.

`    
  def getOptionalPageValue[T](answers: UserAnswers,
    questionPage: QuestionPage[Boolean],
    optionalPage: QuestionPage[T]
  ): EitherNec[ValidationError, Option[T]] =
    getPageValue(questionPage) match {
      case Right(true)  => getPageValue(optionalPage).map(Some(_))
      case Right(false) => unexpectedValueDefined(answers, optionalPage)
      case Left(errors) => Left(errors)
    }`

Reply:
agreed